datasets/hr/public_officials/crawler.py

```diff
@@ -0,0 +1,126 @@
+import csv
+from datetime import datetime
+from typing import Any, Optional
+
+from pantomime.types import CSV
+
+from zavod import Context, Entity
+from zavod import helpers as h
+from zavod.logic.pep import categorise
+
+FIELD_NAMES = [
+    "Ime",
+    "Prezime",
+    "Primarna dužnost",  # first affiliation
+    "Primarna Pravna osoba u kojoj obnaša dužnost",
+    "Primarna Datum početka obnašanja dužnosti",
+    "Primarna Datum kraja obnašanja dužnosti",
+    "Sekundarna dužnost",  # second affiliation
+    "Sekundarna Pravna osoba u kojoj obnaša dužnost",
+    "Sekundarna Datum početka obnašanja dužnosti",
+    "Sekundarna Datum kraja obnašanja dužnosti",
+]
+
+
+def convert_date(date: Optional[str]) -> Optional[str]:
+    return datetime.strptime(date, "%d/%m/%Y").strftime("%Y-%m-%d") if date else None
+
+
+def make_legal_entity(context: Context, legal_entity_name: str) -> Optional[Entity]:
+    legal_entity: Optional[Entity] = None
+    if legal_entity_name:
+        legal_entity = context.make("LegalEntity")
+        legal_entity.id = context.make_slug(legal_entity_name)
+        legal_entity.add("name", legal_entity_name)
+        context.emit(legal_entity)
+    return legal_entity
+
+
+def handle_affiliation_data(context: Context, person: Entity, data: dict) -> None:
+    """Creates, emits Position, Occupancy, and optionally a LegalEntity provided that the Occupancy meets
+    OpenSanctions criteria.
+    Note that the provided data is incomplete and the modeling reflects this:
+      * A position name with no legal entity name results in a position with no Organization reference
+      * A legal entity name with no position name results in an position named "Unspecified position"
+         with organizational reference
+      * A start and/or end date with no position name or legal entity name results in an unspecified position
+         with no organizational reference. This is done for fidelity to the source.
+    """
+    legal_entity_name = data.pop("Pravna osoba u kojoj obnaša dužnost")
+    position_name = data.pop("dužnost")
+    start_date = data.pop("Datum početka obnašanja dužnosti")
+    end_date = data.pop("Datum kraja obnašanja dužnosti")
+    if any([legal_entity_name, position_name, start_date, end_date]):
+        legal_entity = make_legal_entity(context, legal_entity_name)
+        position_name = position_name or "Unspecified position"
+        position = h.make_position(
+            context,
+            position_name,
+            topics=None,
+            organization=legal_entity,
+        )
```

Comment:
We don't currently create entities for the Position:organization property.

What we normally do here is `f"{position_name}, {entity_name}"` if both are available, otherwise just the one.

If neither is available, we skip the position unless we can infer a specific position from the dataset (e.g. we know the source only contains members of parliament)

Sorry for the sidetrack